File: flutter_app/lib/models/item_match.dart
class ItemMatch {
  final String? id;
  final double confidenceScore;
  final double imageSimilarity;
  final double textSimilarity;
  final double categoryMatch;
  final String matchLevel;
  final bool isConfirmed;
  final bool isDismissed;
  final String? createdAt;
  final String? confirmedAt;
  final MatchItem? lostItem;
  final MatchItem? foundItem;

  ItemMatch({
    this.id,
    required this.confidenceScore,
    required this.imageSimilarity,
    required this.textSimilarity,
    required this.categoryMatch,
    required this.matchLevel,
    this.isConfirmed = false,
    this.isDismissed = false,
    this.createdAt,
    this.confirmedAt,
    this.lostItem,
    this.foundItem,
  });

  factory ItemMatch.fromJson(Map<String, dynamic> json) {
    return ItemMatch(
      id: json['id']?.toString(),
      confidenceScore: (json['confidenceScore'] ?? 0).toDouble(),
      imageSimilarity: (json['imageSimilarity'] ?? 0).toDouble(),
      textSimilarity: (json['textSimilarity'] ?? 0).toDouble(),
      categoryMatch: (json['categoryMatch'] ?? 0).toDouble(),
      matchLevel: json['matchLevel'] ?? 'LOW',
      isConfirmed: json['isConfirmed'] ?? false,
      isDismissed: json['isDismissed'] ?? false,
      createdAt: json['createdAt'],
      confirmedAt: json['confirmedAt'],
      lostItem: json['lostItem'] != null
          ? MatchItem.fromJson(json['lostItem'])
          : null,
      foundItem: json['foundItem'] != null
          ? MatchItem.fromJson(json['foundItem'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'confidenceScore': confidenceScore,
      'imageSimilarity': imageSimilarity,
      'textSimilarity': textSimilarity,
      'categoryMatch': categoryMatch,
      'matchLevel': matchLevel,
      'isConfirmed': isConfirmed,
      'isDismissed': isDismissed,
      if (createdAt != null) 'createdAt': createdAt,
      if (confirmedAt != null) 'confirmedAt': confirmedAt,
    };
  }

  String get matchLevelText {
    switch (matchLevel) {
      case 'HIGH':
        return 'High Confidence';
      case 'MEDIUM':
        return 'Medium Confidence';
      default:
        return 'Low Confidence';
    }
  }

  String get confidencePercentage => '${confidenceScore.toStringAsFixed(1)}%';
}

class MatchItem {
  final String? id;
  final String title;
  final String? description;
  final String? category;
  final String? imageUrl;
  final String? lostFoundDate;
  final String? lostFoundLocation;
  final String? reporterName;
  final String? reporterEmail;
  final String? reporterPhoneNo;

  MatchItem({
    this.id,
    required this.title,
    this.description,
    this.category,
    this.imageUrl,
    this.lostFoundDate,
    this.lostFoundLocation,
    this.reporterName,
    this.reporterEmail,
    this.reporterPhoneNo,
  });

  factory MatchItem.fromJson(Map<String, dynamic> json) {
    return MatchItem(
      id: json['id']?.toString(),
      title: json['title'] ?? '',
      description: json['description'],
      category: json['category']?.toString(),
      imageUrl: json['imageUrl'],
      lostFoundDate: json['lostFoundDate']?.toString(),
      lostFoundLocation: json['lostFoundLocation'],
      reporterName: json['reporterName'],
      reporterEmail: json['reporterEmail'],
      reporterPhoneNo: json['reporterPhoneNo'],
    );
  }
}

File: src/main/java/hyk/springframework/lostandfoundsystem/domain/ItemMatch.java
package hyk.springframework.lostandfoundsystem.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import hyk.springframework.lostandfoundsystem.domain.security.User;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;

/**
 * Entity to store match results between lost and found items
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "item_match")
public class ItemMatch extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lost_item_id")
    @JsonIgnoreProperties({ "comments", "user" })
    private LostFoundItem lostItem;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "found_item_id")
    @JsonIgnoreProperties({ "comments", "user" })
    private LostFoundItem foundItem;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lost_user_id")
    @JsonIgnoreProperties({ "lostFoundItems", "roles", "authorities", "password", "confirmedPassword" })
    private User lostItemUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "found_user_id")
    @JsonIgnoreProperties({ "lostFoundItems", "roles", "authorities", "password", "confirmedPassword" })
    private User foundItemUser;

    @Column(name = "confidence_score")
    private Double confidenceScore;

    @Column(name = "image_similarity")
    private Double imageSimilarity;

    @Column(name = "text_similarity")
    private Double textSimilarity;

    @Column(name = "category_match")
    private Double categoryMatch;

    @Column(name = "match_level")
    private String matchLevel; // HIGH, MEDIUM, LOW

    @Column(name = "is_confirmed")
    private Boolean isConfirmed = false;

    @Column(name = "is_dismissed")
    private Boolean isDismissed = false;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "confirmed_at")
    private LocalDateTime confirmedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
